enhancements/ingress/mutable-publishing-scope.md

```diff
@@ -0,0 +1,348 @@
+scope.  Alternatively, if the user reverts the change, the `Progressing` status
+condition changes back to `False`, and the state of the IngressController is as
+it was before the user first changed the value of
+`spec.endpointPublishingStrategy.loadBalancer.scope`.
+
+In addition to deleting the IngressController explicitly, it is possible to
+annotate it with the newly defined
+`ingress.operator.openshift.io/auto-delete-load-balancer` annotation.  If the
+operator observes that this annotation is set and that its
+`spec.endpointPublishingStrategy.loadBalancer.scope` field has been changed, the
+operator automatically deletes the Service if needed to complete a scope-change
+operation.  This purpose of this annotation is to simplify automation: A tool
+can simultaneously update `spec.endpointPublishingStrategy.loadBalancer.scope`
+and set the annotation to instruct the operator to perform the operation
+automatically.  This annotation is not intended for end-users to use directly.
+
+
+### Validation
+
+This enhancement adds no new API fields or values and requires no new API
+validation.
+
+### User Stories
+
+#### As a cluster administrator, I want my default IngressController to be private (internal)
+
+If the user installs a new cluster without specifying that the cluster is
+"private", the default IngressController is created with "External" scope.  This
+enhancement enables the cluster administrator to change the scope to "Internal"
+as follows:
+
+```shell
+oc -n openshift-ingress-operator patch ingresscontrollers/default --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"type":"LoadBalancerService","loadBalancer":{"scope":"Internal"}}}}'
+```
+
+The cluster administrator can check the status of the IngressController as
+usual:
+
+```shell
+oc -n openshift-ingress-operator get ingresscontrollers/default -o yaml
+```
+
+The `Progressing` status condition will indicate whether the cluster
+administrator must take further action.  The status condition may indicate that
+the cluster administrator needs to delete the Service, like so:
+
+```shell
+oc -n openshift-ingress delete services/router-default
+```
+
+If the cluster administrator deletes the Service, the operator then recreates it
+with internal scope.
+
+
+#### As a cluster administrator, I want to make a private (internal) IngressController public (external)
+
+The cluster administrator can change the scope of an IngressController to be
+internal (private) as follows:
+
+```shell
+oc -n openshift-ingress-operator patch ingresscontrollers/private --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"type":"LoadBalancerService","loadBalancer":{"scope":"Internal"}}}}'
+```
+
+Again, the cluster administrator can check the IngressController's status
+conditions and may need to delete a Service to complete the change in scope.
+
+#### As a cluster administrator, I want to cancel a change to scope before it is completed
```

Comment:
I think this use case will be of little value, and though it is easy to implement and describe, it somewhat muddies an otherwise straightforward documentation.

Reply:
It's important because we've had support cases where users changed the scope on the IngressController and then realized that completing the change would cause the service load-balancer's IP address to change and wanted to back out.